### PR TITLE
CPM-1084: Update ES Identifier filters to use main identifier attribute

### PR DESCRIPTION
**Pull request by Balamung**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Bundle/Resources/config/queries.yml [hunk 1]
```diff
@@ -222,3 +222,14 @@ services:
         class: Akeneo\Pim\Structure\Bundle\Query\PublicApi\Family\Sql\SqlGetFamilies
         arguments:
             - '@database_connection'
+
+    akeneo.pim.structure.query.sql_get_main_identifier_attribute_code:
+        class: Akeneo\Pim\Structure\Bundle\Query\PublicApi\Attribute\Sql\SqlGetMainIdentifierAttributeCode
+        arguments:
+            - '@database_connection'
+
+    Akeneo\Pim\Structure\Component\Query\PublicApi\Attribute\GetMainIdentifierAttributeCode:
```
Comment (Balamung) [inline src/Akeneo/Pim/Structure/Bundle/Resources/config/queries.yml:231]: double

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Filter/Field/IdentifierFilter.php [hunk 2]
```diff
@@ -70,112 +71,294 @@ protected function checkValue($property, $operator, $value)
...
+            'bool' => [
+                'should' => [
+                    [
+                        'query_string' => [
+                            'default_field' => $productIdentifierField,
+                            'query'         => $searchString,
+                        ],
+                    ],
+                    [
+                        'query_string' => [
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Filter/Field/IdentifierFilter.php:227]: match

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Filter/Field/IdentifierFilter.php [hunk 3]
```diff
@@ -70,112 +71,294 @@ protected function checkValue($property, $operator, $value)
...
+    private function buildIdentifierSearchFilter(
+        string $searchString,
+        string $productDocumentType,
+        string $productIdentifierField,
+        string $productModelDocumentType,
+        string $productModelIdentifierField
+    ): array {
+        $productClause = [
+            'bool' => [
+                'should' => [
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Filter/Field/IdentifierFilter.php:219]: filter / must

File: tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/Filter/Field/IdentifierFilterSpec.php [hunk 4]
```diff
@@ -235,7 +199,8 @@ function it_throws_an_exception_when_the_given_value_is_not_a_string_with_unsupp
 
     function it_throws_an_exception_when_the_given_value_is_null(
         SearchQueryBuilder $sqb
-    ) {
+    )
+    {
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/Filter/Field/IdentifierFilterSpec.php:203]: nope :)

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Filter/Field/IdentifierFilter.php [hunk 5]
```diff
@@ -70,112 +71,282 @@ protected function checkValue($property, $operator, $value)
...
+                ));
+                $this->searchQueryBuilder->addFilter($this->buildFieldShouldExistClause(
+                    $productDocumentType,
+                    $productIdentifierField,
+                    $productModelDocumentType,
+                    $productModelIdentifierField
+                ));
                 break;
 
             case Operators::EQUALS:
```
Comment (Balamung) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Filter/Field/IdentifierFilter.php:131]: Test with term instead of query string

**Review by Beninho — APPROVED** [review]

**Review by cemorin — APPROVED** [review]

**Pull request merged**